### PR TITLE
updated links

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Link to sample test page:
Related PRs/issues: #11428 

This PR updates some URLs on the donate pages per request of the donor care team, including:

**Donate Landing Page:**
- Other ways to give: [Check](https://foundation.mozilla.org/ways-to-give#check) >>  update to "Other ways to give: Check, Bank Transfer, Stocks" and update link to `/donate/ways-to-give`.
- Question donating? Visit our [FAQ](https://foundation.mozilla.org/faq) for answers to most common questions. >> Update link to `donate/help/#frequently-asked-questions`
- Need to reach us about your donation? [Contact us](https://foundation.mozilla.org/help/).>>  Update link to `/donate/help`

**Donate Help Page:**
- "If you need help with a [donation](foundation.mozilla.org/?form=donate) to the Mozilla Foundation, please select the reason for your inquiry from the drop down and a donor care representative will get back to you as soon as possible."  >> Update link to a relative URL, that way the fundraise up modal opens on the same page instead of opening the foundation homepage on a new tab.

### CODE REVIEW

File: network-api/networkapi/templates/donate/fragments/formassembly_body.html [hunk 1]
```diff
@@ -89,7 +89,7 @@
                     donations via direct debit in Euros through Single European Payment Area
                     (SEPA) transfer.
                 {% endblocktrans %}
-            </span><br><br><div><b><a target="_blank" href="https://foundation.mozilla.org/?form=donate&amp;utm_medium=web&amp;utm_source=faq-page&amp;utm_campaign=23-Donor-Care&amp;utm_term=en"><u>{% trans "Make your donation online now" %}<br><br></u></a></b><i>{% trans "Still have questions? Contact us by submitting a case." %}</i><br></div></div></div></div>
+            </span><br><br><div><b><a href="?form=donate&amp;utm_medium=web&amp;utm_source=faq-page&amp;utm_campaign=23-Donor-Care&amp;utm_term=en"><u>{% trans "Make your donation online now" %}<br><br></u></a></b><i>{% trans "Still have questions? Contact us by submitting a case." %}</i><br></div></div></div></div>
```
Comment (danielfmiranda) [inline network-api/networkapi/templates/donate/fragments/formassembly_body.html:92]: Updated this in form assembly to be a relative URL, that way, the donate form will open without leading the users to the foundation homepage on another tab!

Also updated revision number on both this file and `formassembly_head.html` to reflect new version number

**Review by mtdenton — APPROVED** [review]
LGTM!

File: network-api/networkapi/templates/donate/fragments/formassembly_body.html (hunk 1 above)
Comment (mmmavis) [reply network-api/networkapi/templates/donate/fragments/formassembly_body.html:92]: Hey @danielfmiranda , jsut wanted to confirm that it is okay to hardcoded`utm_term`  to `en`

Comment (danielfmiranda) [reply network-api/networkapi/templates/donate/fragments/formassembly_body.html]: Thanks for the catch @mmmavis! After confirming with @miaatmozilla that this should be removed, I have updated the link in form assembly and also updated this file to remove `utm_term=en`

**Review by mmmavis — APPROVED** [review]
@danielfmiranda thanks for double checking with the stakeholders and updating the PR accordingly. R+!

**Pull request merged**